pre_build/cccp_index_pre_build_check.py

```diff
@@ -0,0 +1,162 @@
+        new_project[0]['project']['rundotshargs'] = ''
+
+    new_project[0]['project']['desired_tag'] = desiredtag
+    new_project[0]['project']['prebuild_script'] = prebuild_script
+    return new_project
+
+
+def get_projects_from_index(indexdlocation):
+    projects = []
+    for yamlfile in glob(indexdlocation + "/*.yml"):
+        if "index_template" not in yamlfile:
+            stream = open(yamlfile, 'r')
+            index_yaml = yaml.load(stream, Loader=yaml.BaseLoader)
+            stream.close()
+            # print index_yaml['Projects']
+
+            for project in index_yaml['Projects']:
+                if(project['git-url'] is None):
+                    continue
+                else:
+                    try:
+                        appid = project['app-id']
+                        jobid = project['job-id']
+                        giturl = project['git-url']
+                        gitpath = project['git-path'] \
+                            if (project['git-path'] is not None) else ''
+                        gitbranch = project['git-branch']
+                        try:
+                            desiredtag = project['desired-tag'] \
+                                if (project['desired-tag'] is not None) \
+                                else 'latest'
+                        except Exception:
+                            desiredtag = 'latest'
+
+                        desiredtag = str(desiredtag)
+
+                        new_proj = [{'project': {}}]
+
+                        appid = appid.replace(
+                            '_', '-').replace('/', '-').replace('.', '-')
+                        jobid = jobid.replace(
+                            '_', '-').replace('/', '-').replace('.', '-')
+
+                        # Set the pre-build data if available
+                        prebuild_script = 'none' if not project.get(
+                            'prebuild-script') else project['prebuild-script']
+
+                        # Check for the project only if it requires pre-build
+                        if(prebuild_script != 'none'):
+                            projects.append(
+                                projectify(
+                                    new_proj, appid, jobid, giturl, gitpath,
+                                    gitbranch, desiredtag,
+                                    prebuild_script)
+                            )
+                    except Exception as e:
+                        print("Failed to projectify %s" %
+                              str(project))
+                        print("Error is : %s %s" % (str(e), sys.exc_info()[0]))
+                        raise
+    return projects
+
+
+def run_command(command):
+    """
+    runs the given shell command using subprocess
+    """
+    proc = subprocess.Popen(command,
+                            stdout=subprocess.PIPE)
+
+    return proc.communicate()
+
+
+def main(indexdlocation):
```

Comment:
I know I sound like a pain but, can we have more documentation in form of comments here? I'm asking because most code in this PR and specially in this file is not going to be easy to test through CI, DevCloud or pre-prod. It's directly going to go into prod.

Reply:
added.